### PR TITLE
early postgres connection

### LINKED ISSUE #6: Connect to PostgreSQL before starting the bot
In the interest of being able to access `self.pool` in `Cog.__init__`, it would be nice to establish the PostgreSQL connection early.  (We get to fail early, too.)

### PR DESCRIPTION
**Pull request by lun-4**
closes #6

### CODE REVIEW

**Review by slice — CHANGES_REQUESTED** [review]
I don't think that the connection should be done in the constructor. Calling `Bot(...)` and having it automatically attempt to spin up a connection would be disruptive, in my opinion. I think it would be better to call `_postgres_connect` in `lifesaver.cli`.

**Comment by lun-4** [comment]
But isn't bot the one doing the side-effects of configuration reading? locking it to `lifesaver.cli` only doesn't sound like a good idea..

**Comment by slice** [comment]
> But isn't bot the one doing the side-effects of configuration reading?

No? It receives the `BotConfig` object in `__init__`, it doesn't read from the filesystem or anything. Besides, I think connecting to Postgres in a constructor is out of place and pretty drastic.

> locking it to `lifesaver.cli` only doesn't sound like a good idea..

Can you elaborate on that?

**Comment by lun-4** [comment]
Consider some user of lifesaver that just wants to run a script to start the bot instead of `lifesaver.cli`, but that's likely an edge-case that can be ignored.

**Comment by slice** [comment]
Yes, users should be using `lifesaver.cli`. I can provide a method to kickstart all of the required effects that `lifesaver.cli` can use if they really want to use a start script.

File: lifesaver/cli.py [hunk 1]
```diff
@@ -17,6 +17,17 @@ def resolve_class(specifier: str):
     return loaded_class
 
 
+async def _postgres_connect(bot):
```
Comment (slice) [inline lifesaver/cli.py]: The method should be kept on the bot as that's where the database lives. It should still be called from `lifesaver.cli` though.

**Review by slice — CHANGES_REQUESTED** [review]

**Pull request merged**